core/build.gradle

```diff
@@ -61,9 +61,11 @@ import com.microsoft.applicationinsights.build.tasks.PropsFileGen
 
 archivesBaseName = 'applicationinsights-core'
 
+apply from: "$buildScriptsDir/provided-configuration.gradle"
+
 dependencies {
-    compileOnly (project(':agent')) { transitive = false }
-    compileOnly ([group: 'org.glowroot.instrumentation', name: 'instrumentation-api', version: '0.14.9']) { transitive = false }
+    optional (project(':agent')) { transitive = false }
+    provided ([group: 'org.glowroot.instrumentation', name: 'instrumentation-api', version: '0.14.9']) { transitive = false }
```

Comment:
Now that I look at this again, I think the configuration for glowroot should match `:agent`.